### PR TITLE
Added BitField support

### PR DESCRIPTION
**Pull request by RReverser**
Added recursive bitfields support with documentation and test updates; only left-to-right (like in Big-Endian) mode is supported for now.

### CODE REVIEW

File: src/jparser.js [hunk 1]
```diff
@@ -77,6 +77,56 @@ jParser.prototype.structure = {
...
+				if (bitSize > 0) {
+					fieldValue = ((this.view.getUint8() >>> (8 - (bitShift + bitSize))) & ~(-1 << bitSize)) | (fieldValue << bitSize);
+					bitShift = bitShift + bitSize - 8; // passing negative value for next pass
+				}
+			}
+
+			output[key] = fieldValue;
+		}
+
+        this.current = current;
```
Comment (vjeux) [inline src/jparser.js:127]: I shouldn't have used tabs for indentation :( Can you fix when you used space instead of tabs?

**Comment by vjeux** [comment]
Thanks a lot, this is truly awesome. Very small and yet powerful :)

Can you just fix the space/tab issue and I'm going to merge it. (That's too bad github doesn't provide the ability to accept a pull request but let the author do some final changes)

**Comment by RReverser** [comment]
Sorry, no problems when writing in Sublime, but always too lazy to setup WebStorm for same indentation :)
Will post it a bit earlier.
(Actually, you can accept pull request and commit own changes over it).

**Comment by RReverser** [comment]
You know what... I actually got new idea today, let me close this pull request and submit a bit better implementation later.

**Comment by vjeux** [comment]
Nice, can't wait to see it

**Pull request closed without merging**